src/main/java/com/example/onlinebookstore/repository/BookRepository.java

```diff
@@ -1,13 +1,18 @@
 package com.example.onlinebookstore.repository;
 
 import com.example.onlinebookstore.model.Book;
-import java.util.List;
-import java.util.Optional;
+import java.math.BigDecimal;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> getAll();
-
-    Optional<Book> findById(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
+    @Modifying
+    @Transactional
+    @Query("UPDATE Book b SET b.author = :author, b.title = :title, "
+            + "b.price = :price, b.description = :description, b.coverImage = :coverImage "
+            + "WHERE b.id = :id")
```

Comment:
to update you can just call .save() on existing entity (with id), no need for this method. You can use mapper to create entity from DTO

Reply:
> to update you can just call .save() on existing entity (with id), no need for this method. You can use mapper to create entity from DTO

but it's good choice as well

Reply:
I thought It's more explicit approach. Is it ?